sycl/include/sycl/detail/kernel_desc.hpp

```diff
@@ -151,6 +168,98 @@ template <class KernelNameType> struct KernelInfo {
+#else
+  return KernelInfo<KernelNameType>::getNumParams();
+#endif
+}
+
+template <typename KernelNameType>
+std::vector<kernel_param_desc_t> getKernelParamDescs() {
+  std::vector<kernel_param_desc_t> Result;
+  int NumParams = getKernelNumParams<KernelNameType>();
+  Result.reserve(NumParams);
+  for (int I = 0; I < NumParams; ++I) {
+#ifndef __INTEL_SYCL_USE_INTEGRATION_HEADERS
+    kernel_param_desc_t ParamDesc;
+    ParamDesc.kind =
+        __builtin_sycl_kernel_param_kind(KernelIdentity<KernelNameType>(), I);
+    ParamDesc.info = ParamDesc.kind == kernel_param_kind_t::kind_accessor
+                         ? __builtin_sycl_kernel_param_access_target(
+                               KernelIdentity<KernelNameType>(), I)
+                         : __builtin_sycl_kernel_param_size(
+                               KernelIdentity<KernelNameType>(), I);
+    ParamDesc.offset =
+        __builtin_sycl_kernel_param_offset(KernelIdentity<KernelNameType>(), I);
+    Result.push_back(ParamDesc);
+#else
+    Result.push_back(KernelInfo<KernelNameType>::getParamDesc(I));
+#endif
+  }
+  return Result;
+}
+
+template <typename KernelNameType> constexpr const char *getKernelName() {
+#ifndef __INTEL_SYCL_USE_INTEGRATION_HEADERS
+  return __builtin_sycl_kernel_name(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getName();
+#endif
+}
+
+template <typename KernelNameType> constexpr bool isKernelESIMD() {
+  // TODO Needs a builtin counterpart
+  return KernelInfo<KernelNameType>::isESIMD();
+}
+
+template <typename KernelNameType> constexpr const char *getKernelFileName() {
+#ifndef __INTEL_SYCL_USE_INTEGRATION_HEADERS
+  return __builtin_sycl_kernel_file_name(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getFileName();
+#endif
+}
+
+template <typename KernelNameType>
+constexpr const char *getKernelFunctionName() {
+#ifndef __INTEL_SYCL_USE_INTEGRATION_HEADERS
+  return __builtin_sycl_kernel_function_name(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getFunctionName();
+#endif
+}
+
+template <typename KernelNameType> constexpr unsigned getKernelLineNumber() {
+#ifndef __INTEL_SYCL_USE_INTEGRATION_HEADERS
+  return __builtin_sycl_kernel_line_number(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getLineNumber();
+#endif
+}
+
+template <typename KernelNameType> constexpr unsigned getKernelColumnNumber() {
+#ifndef __INTEL_SYCL_USE_INTEGRATION_HEADERS
+  return __builtin_sycl_kernel_column_number(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getColumnNumber();
+#endif
+}
+
+template <typename KernelNameType> constexpr int64_t getKernelSize() {
+  // TODO needs a builtin counterpart, but is currently only used for checking
+  // cases with external host compiler, which use integration headers.
+  return KernelInfo<KernelNameType>::getKernelSize();
```

Comment:
Same here. Wouldn't this throw an error when `getKernelSize()` is called in non-integration header mode? Since we don't plan to implement this, and eventually plan to replace the `static_assert `which uses this to a FE diagnostic, does it make sense to have this return a dummy value and guard the static assert for integration header mode?

Reply:
And the static assert is already guarded.